tools/deploy_metrics_and_monitoring_resources.sh

```diff
@@ -0,0 +1,155 @@
+echo "...Prometheus operator ready"
+
+rm $DIR/metrics/prometheus-operator-deploy.yaml
+
+kubectl create secret generic additional-scrape-configs --from-file=$DIR/metrics/examples/prometheus/additional-properties/prometheus-additional.yaml
+kubectl create secret generic alertmanager-alertmanager --from-file=alertmanager.yaml=$DIR/metrics/examples/prometheus/alertmanager-config/alert-manager-config.yaml
+
+
+if [[ "$OSTYPE" == "linux-gnu" ]]; then
+    $SED "s/namespace: .*/namespace: $NAMESPACE/" $DIR/metrics/examples/prometheus/install/prometheus.yaml > $DIR/metrics/examples/prometheus/install/prometheus-deploy.yaml
+elif [[ "$OSTYPE" == "darwin"* ]]; then
+    $SED '' "s/namespace: .*/namespace: $NAMESPACE/" $DIR/metrics/examples/prometheus/install/prometheus.yaml > $DIR/metrics/examples/prometheus/install/prometheus-deploy.yaml
+fi
+
+kubectl apply -f $DIR/metrics/examples/prometheus/install/prometheus-rules.yaml -n $NAMESPACE
+kubectl apply -f $DIR/metrics/examples/prometheus/install/prometheus-deploy.yaml -n $NAMESPACE
+rm $DIR/metrics/examples/prometheus/install/prometheus-deploy.yaml
+kubectl apply -f $DIR/metrics/examples/prometheus/install/strimzi-service-monitor.yaml -n $NAMESPACE
+kubectl apply -f $DIR/metrics/examples/prometheus/install/alert-manager.yaml -n $NAMESPACE
+
+# Delay for allowing Prometheus operator to create the Alert manager statefulset
+sleep 15
+
+amReplicas=$(kubectl get statefulsets alertmanager-alertmanager -o jsonpath="{.spec.replicas}" -n $NAMESPACE)
+echo "Waiting for Alert manager to be ready..."
+readyReplicas="0"
+while [ "$readyReplicas" != "$amReplicas" ]
+do
+    sleep 2
+    readyReplicas=$(kubectl get statefulsets alertmanager-alertmanager -o jsonpath="{.status.readyReplicas}" -n $NAMESPACE)
+done
+echo "...Alert manager ready"
+
+# Delay for allowing Prometheus operator to create the Prometheus statefulset
+sleep 15
+
+pReplicas=$(kubectl get statefulsets prometheus-prometheus -o jsonpath="{.spec.replicas}" -n $NAMESPACE)
+echo "Waiting for Prometheus to be ready..."
+readyReplicas="0"
+while [ "$readyReplicas" != "$pReplicas" ]
+do
+    sleep 2
+    readyReplicas=$(kubectl get statefulsets prometheus-prometheus -o jsonpath="{.status.readyReplicas}" -n $NAMESPACE)
+done
+echo "...Prometheus ready"
+
+rm $DIR/metrics/prometheus-operator.yaml
+
+# Grafana
+kubectl apply -f $DIR/metrics/examples/grafana/grafana.yaml -n $NAMESPACE
+
+echo "Waiting for Grafana server to be ready..."
+kubectl rollout status deployment/grafana -w -n $NAMESPACE
+echo "...Grafana server ready"
+
+kubectl expose service/grafana -n $NAMESPACE
+
+sleep 5
+
+# Posting Prometheus datasource and dashboards to Grafana
+curl -X POST http://admin:admin@$(kubectl get routes grafana -o jsonpath='{.status.ingress[0].host}{"\n"}')/api/datasources  -H 'Content-Type: application/json;charset=UTF-8' --data-binary '{"name":"Prometheus","isDefault":true ,"type":"prometheus","url":"http://prometheus:9090","access":"proxy","basicAuth":false}'
```

Comment:
Do we have to do it this way? Can't we load it from config maps like when Prometheus is deployed directly?

Reply:
I am trying to use just the resources from this repo. I think it is possible to use some kind of CM but it has to be stored in some external repo. What do you prefer?

Reply:
Why does it have to be stored in external repo? You can just create it on the fly, or?